protocol/messenger_builder_test.go

```diff
@@ -27,7 +29,8 @@ type testMessengerConfig struct {
 	unhandledMessagesTracker *unhandledMessagesTracker
 	messagesOrderController  *MessagesOrderController
 
-	extraOptions []Option
+	extraOptions   []Option
+	createSettings bool
```

Comment:
Maybe instead of polluting config with a boolean we could create `newTestSettings` and populate `extraOptions` on `testMessengerConfig` creation, as follows:

```go
func newTestSettings() settings.Settings {
    return settings.Settings{
        DisplayName:               DefaultProfileDisplayName,
        ProfilePicturesShowTo:     1,
        ProfilePicturesVisibility: 1,
        URLUnfurlingMode:          settings.URLUnfurlingAlwaysAsk,
    }
}
```

```go
m, err := newTestMessenger(s.shh, testMessengerConfig{
    logger:         s.logger.Named(fmt.Sprintf("messenger-%s", name)),
    name:           name,
    extraOptions:   []Option{WithAppSettings(newTestSettings(), params.NodeConfig{})},
})
```

Reply:
Actually this was the first thing I did. 
But then I tried to actually set `DisplayName` to the real display name, and this made sense to fill automatically inside the config. And this somehow didn't work, so I left it to be `DefaultProfileDisplayName`.

Will do as you say here, agree 👍